### PR TITLE
Aesthetic changes to notif info generator

### PR DESCRIPTION
**Pull request by ryanpbrewster**
@andrewconner two changes, one aesthetic and one efficiency

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/common/Implicits.scala [hunk 1]
```diff
@@ -77,6 +77,17 @@ final class IterableExtensionOps[A, Repr](xs: IterableLike[A, Repr]) {
...
+  def flatAugmentWith[Acc, B, That](fn: A => Option[B])(implicit cbf: CanBuildFrom[Repr, (A, B), That]): That = {
+    val builder = cbf(xs.repr)
+    xs.foreach { x =>
+      fn(x).foreach { y => builder += x -> y }
+    }
+    builder.result()
+  }
+  def augmentWith[Acc, B, That](fn: A => B)(implicit cbf: CanBuildFrom[Repr, (A, B), That]): That = {
+    flatAugmentWith(x => Option(fn(x)))
+  }
```
Comment (ryanpbrewster) [inline kifi-backend/modules/common/app/com/keepit/common/Implicits.scala:90]: Not convinced about the naming. This is for situations where you do things like

```
xs.map(x => x -> fn(x)).toMap
```

or, even worse

```
xs.flatMap(x => fnOpt(x).map(x -> _)).toMap
```

File: kifi-backend/modules/eliza/app/com/keepit/notify/info/NotificationInfoGenerator.scala [hunk 2]
```diff
@@ -66,9 +70,12 @@ class NotificationInfoGenerator @Inject() (
     val orgsF = shoeboxServiceClient.getBasicOrganizationsByIds(orgRequests)
     val keepsF = shoeboxServiceClient.getBasicKeepsByIds(keepRequests)
     val summaryF = keepsF.flatMap { keeps =>
-      val uriToKeepIds = keeps.map { case (id, bk) => id -> bk.uriId }.groupBy(_._2).mapValues(_.keySet)
-        .flatMap { case (id, ks) => NormalizedURI.decodePublicId(id).toOption.map(_ -> ks) }
-      roverServiceClient.getUriSummaryByUris(uriToKeepIds.keys.toSet).map { s => s.flatMap { case (uriId, summary) => uriToKeepIds(uriId).map(_ -> summary) } }
+      val uriIdByKeep = summaryRequests.flatAugmentWith { kId =>
```
Comment (ryanpbrewster) [inline kifi-backend/modules/eliza/app/com/keepit/notify/info/NotificationInfoGenerator.scala:73]: We should only be grabbing the uris from keeps with a uri summary request

File: kifi-backend/modules/common/app/com/keepit/common/Implicits.scala (hunk 1 above)
Comment (ryanpbrewster) [inline kifi-backend/modules/common/app/com/keepit/common/Implicits.scala:90]: `xs.augmentWith(fn).toMap` or `xs.flatAugmentWith(fnOpt).toMap`

File: kifi-backend/modules/eliza/app/com/keepit/notify/info/NotificationInfoGenerator.scala (hunk 2 above)
Comment (andrewconner) [inline kifi-backend/modules/eliza/app/com/keepit/notify/info/NotificationInfoGenerator.scala:73]: Ah you're right. I think I had it, and messed that up upon rewrites.

**Comment by andrewconner** [comment]
We're going to end up with our own Scala collections API one of these days. If you like it better, merge hoy.

**Pull request merged**